be/src/io/cache/block/block_file_cache.cpp

```diff
@@ -51,16 +51,34 @@ IFileCache::Key IFileCache::hash(const std::string& path) {
     return Key(key);
 }
 
-std::string IFileCache::get_path_in_local_cache(const Key& key, size_t offset,
-                                                bool is_persistent) const {
-    auto key_str = key.to_string();
-    std::string suffix = is_persistent ? "_persistent" : "";
-    if constexpr (USE_CACHE_VERSION2) {
-        return fs::path(_cache_base_path) / key_str.substr(0, KEY_PREFIX_LENGTH) / key_str /
-               (std::to_string(offset) + suffix);
-    } else {
-        return fs::path(_cache_base_path) / key_str / (std::to_string(offset) + suffix);
+std::string IFileCache::cache_type_to_string(CacheType type) {
+    switch (type) {
+    case CacheType::INDEX:
+        return "_idx";
+    case CacheType::DISPOSABLE:
+        return "_disposable";
+    case CacheType::NORMAL:
+        return "";
+    }
+    return "";
+}
+
+CacheType IFileCache::string_to_cache_type(const std::string& str) {
```

Comment:
Why does only match the first character?

Reply:
Only the first character is needed to distinguish the type. It is faster than comparing with string.